File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-4.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        l = 0
        r = 0
        se = set()
        le = 0

        while r<len(s):
            while s[r] in se:
                se.remove(s[l])
                l+=1
            se.add(s[r])
            le = max(le, r-l+1)
            r+=1

        return le

        

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-4.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int l = 0;
        int res = 0;

        for (int r=0; r<prices.size(); r++){
            if (prices[r] > prices[l]){
                int prof = prices[r] - prices[l];
                res = std::max(res, prof);
            }
            else{l=r;}
        }

        return res;
    }
};
